<!--
  Copyright Clarisoft, a Modus Create Company, 20/07/2023, licensed under the
  EUPL-1.2 or later. This open-source code is licensed following the Attribution
  4.0 International (CC BY 4.0) - Creative Commons — Attribution 4.0 International
  — CC BY 4.0.

  Following this, you are accessible to:

  Share - copy and redistribute the material in any medium or format.
  Adapt - remix, transform, and build upon the material commercially.

  Remark: The licensor cannot revoke these freedoms if you follow the license
  terms.

  Under the following terms:

  Attribution - You must give appropriate credit, provide a link to the license,
  and indicate if changes were made. You may do so reasonably but not in any way
  that suggests the licensor endorses you or your use.
  No additional restrictions - You may not apply legal terms or technological
  measures that legally restrict others from doing anything the license permits.
-->
<div class="variants-of-concern">

  <mat-card>
    <mat-card-title>
      <div class="row">
        <div class="col-md-12 text-center text-sm-left">
          Variants of Concern
        </div>
      </div>
    </mat-card-title>
    <br/>
    <mat-card-subtitle>
      <div class="row">
        <div class="col-md-7 mb-4 text-center text-sm-left">
          <p>For these variants, clear evidence is available indicating a significant impact on
            transmissibility, severity
            and/ or immunity that is likely to have an impact on the epidemiological situation in the EU/
            EEA. The combined genomic,
            epidemiological, and in-vitro evidence for these properties invokes at least moderate
            confidence. In addition,
            all the criteria for variants of interest and under monitoring outlined below apply.
          </p>
        </div>
        <div class="col-md-5 mb-4 text-center text-sm-left text-md-right">
          <app-sources
            [sourceType]="SourceType.GRAPH"
            [sources]="sources">
          </app-sources>
        </div>
      </div>
    </mat-card-subtitle>
    <mat-card-content [ngStyle]="{'display': isCollapsed === true ? 'none' :'block'}">
      <div class="mb-5">
        <mat-tab-group #tabs (click)="tabClick()" (selectedTabChange)="changeVariant($event)"
                       [(selectedIndex)]="activeTabIndex" animationDuration="0ms">
          <mat-tab>
            <table aria-label="Variants of Concern table info">
              <thead>
              <th> Lineage + additional mutations</th>
              <th> Country first detected (community)</th>
              <th> Spike mutations of interest</th>
              <th> Year and month of first detection</th>
              <th> Impact on transmissibility</th>
              <th> Impact on immunity</th>
              <th> Impact on severity</th>
              <th> Transmission in EU/ EEA</th>
              </thead>
              <tbody>
              <tr>
                <td>
                  The variant designation specified by one or more Pango lineages and any additional
                  characteristic spike protein changes. An alternative description may be used if the
                  variant is not easy to describe using this nomenclature. Each lineage in the table
                  is linked to the respective lineage page of the Pango lineages website.
                </td>
                <td>
                  Only present if there is moderate confidence in the evidence relating to the first
                  country of detection.
                </td>
                <td>
                  Not all spike protein amino acid changes are included - this is not a full reference
                  of assignment of the variants. It includes changes to spike protein residues 319-541
                  (receptor binding domain) and 613-705 (the S1 part of the S1/ S2 junction and a
                  small stretch on the S2 side), and any additional unusual changes specific to the
                  variant
                </td>
                <td>
                  As reported in the GISAID EpiCoV database. This can be adjusted backwards in time if
                  new retrospective detections are made.
                </td>
                <td colspan="3">
                  <div>Each category is annotated as <strong>increased</strong>,
                    <strong>reduced</strong>, <strong>similar</strong>,
                    <strong>unclear</strong>,
                    or <strong>no evidence depending on the currently available
                      evidence.</strong><br>
                    <ul style="list-style-position: inside;">
                      <li>
                        <strong>Increased</strong> or <strong>reduced</strong> means that there
                        is evidence
                        demonstrating that the property is different enough for the
                        variant compared to the previously circulating variants that it is
                        likely to
                        have an impact on the epidemiological situation in the EU/ EEA
                      </li>
                      <li>
                        <strong>Similar</strong> means that there is evidence that demonstrates
                        that the
                        property is not different enough
                        for this variant compared to previously circulating variants that it is
                        unlikely to have an impact.
                      </li>
                      <li>
                        <strong>Unclear</strong> means that the current evidence is preliminary
                        or
                        contradictory enough to make the assessment uncertain.
                      </li>
                      <li>
                        <strong>No evidence</strong> means that no evidence has yet been
                        evaluated for
                        this
                        category.
                      </li>
                      <li>
                        The evidence is further annotated with <strong>v</strong> or
                        <strong>m</strong> to indicate
                        whether the evidence is available for the variant itself
                        <strong>(v)</strong> or
                        for
                        mutations associated with the variant <strong>(m)</strong>.
                      </li>
                    </ul>
                  </div>
                </td>
                <td>
                  Categorised as <strong>dominant</strong>, <strong>community</strong>, <strong>outbreak(s)</strong>,
                  and <strong>sporadic/
                  travel</strong>.
                  The categories are qualitative, and the assessment is based on surveillance data
                  collected in TESSy, GISAID EpiCoV data,
                  epidemic intelligence data, and direct communications with the affected countries.
                </td>
              </tr>

              </tbody>
            </table>

          </mat-tab>
          <mat-tab *ngFor="let variant of variants" label="{{variant.name | uppercase}}">
            <table [attr.aria-label]="variant.name + ' data'" class="variants-of-concern__data-table">
              <thead>
              <th> Lineage + additional mutations</th>
              <th> Country first detected (community)</th>
              <th> Spike mutations of interest</th>
              <th> Year and month of first detection</th>
              <th> Impact on transmissibility</th>
              <th> Impact on immunity</th>
              <th> Impact on severity</th>
              <th> Transmission in EU/ EEA</th>
              </thead>
              <tbody>
              <tr>
                <td>
                  <a href='https://cov-lineages.org/global_report_{{variant.lineage}}.html' style="text-decoration: underline; color: black;"
                     target='_blank'>{{variant.lineage}}</a>
                </td>
                <td>
                  {{variant.country_first_detected}}
                </td>
                <td>
                  {{variant.spike_mutations}}
                </td>
                <td>
                  {{moment(variant.date_first_detection).format('MMM YYYY')}}
                </td>
                <td>
                  {{variant.impact_transmissibility}}
                </td>
                <td>
                  {{variant.impact_immunity}}
                </td>
                <td>
                  {{variant.impact_severity}}
                </td>
                <td>
                  {{variant.transmission_eu_eea}}
                </td>
              </tr>
              </tbody>
            </table>
          </mat-tab>
        </mat-tab-group>
      </div>
      <div class="row">
        <div class="col-lg-7">
          <app-variants-of-concern-tree
            #tree>
          </app-variants-of-concern-tree>
        </div>
        <div class="col-lg-5">
          <mat-card class="world-map">
            <app-map
              #map
              [filters]="geographicalAreaFilters"
              [hasCustomTitle]="true"
              [hideInputs]=true
              [isZoomDisabled]="true"
              [mapHeight]=800
              [module]="'variants'"
              [removeLegends]="true"
              [selectedFilterValue]="currentVariant"
            ></app-map>
          </mat-card>
        </div>
      </div>
      <div *ngIf="currentVariant !== 'None'">
        <div *ngIf="isLoading()"
             class="mat-spinner-container loading-chart-placeholder">
          <mat-spinner></mat-spinner>
        </div>
        <div *ngIf="isLoaded()" class="row">
          <div *ngIf="dailyChart" class="col-lg-12">
            <mat-card>
              <app-highcharts [chartType]="chartType" [hideLegend]="true"
                              [proportionChart]="true" [series]="dailySeries"
                              [showExportMenu]="true"
                              [stacking]="true" [xAxisData]="dailyChart.total.xAxis"

              ></app-highcharts>
            </mat-card>
          </div>
        </div>
      </div>
    </mat-card-content>
  </mat-card>
</div>
