<!--
  Copyright Clarisoft, a Modus Create Company, 20/07/2023, licensed under the
  EUPL-1.2 or later. This open-source code is licensed following the Attribution
  4.0 International (CC BY 4.0) - Creative Commons — Attribution 4.0 International
  — CC BY 4.0.

  Following this, you are accessible to:

  Share - copy and redistribute the material in any medium or format.
  Adapt - remix, transform, and build upon the material commercially.

  Remark: The licensor cannot revoke these freedoms if you follow the license
  terms.

  Under the following terms:

  Attribution - You must give appropriate credit, provide a link to the license,
  and indicate if changes were made. You may do so reasonably but not in any way
  that suggests the licensor endorses you or your use.
  No additional restrictions - You may not apply legal terms or technological
  measures that legally restrict others from doing anything the license permits.
-->
<mat-card class="landing-card summary-section">
  <div *ngIf="isLoading()" class="mat-spinner-container">
    <mat-spinner></mat-spinner>
  </div>
  <div *ngIf="isLoaded()">
    <div class="summary-section__name-icon-container">
      <div>Genetic Variation</div>
      <div class="card-actions">
        <mat-icon>drag_indicator</mat-icon>
        <mat-icon (click)="hide()">visibility_off</mat-icon>
        <mat-icon (click)="showInfo = !showInfo">info</mat-icon>
      </div>
    </div>
    <mat-card-title class="text-center text-sm-left">HTS (High Throughput Sequencing)</mat-card-title>

    <mat-card-subtitle class="text-center text-sm-left">
      <app-sources
        [sourceType]="SourceType.SUMMARY"
        [sources]="sources">
      </app-sources>
    </mat-card-subtitle>

    <mat-card-content>
      <p *ngIf="showInfo" class="info-text">
        For these variants, clear evidence is available indicating a significant impact
        on transmissibility, severity and/ or immunity that is likely to have an impact
        on the epidemiological situation in the EU/ EEA. The combined genomic,
        epidemiological, and in-vitro evidence for these properties invokes at least
        moderate confidence. In addition, all the criteria for variants of interest and
        under monitoring outlined below apply.
      </p>
      <mat-list>
        <ng-container *ngIf="variants.length">
          <div *ngFor="let variant of variants, let i = index">
            <mat-list-item *ngIf="i < 4">
              <div class="col-xl-9 label">
                {{variant.name}}
              </div>
              <div class="col-xl-3 value text-right">
                {{variant.proportion_of_sequences | number: defaultNumberFormat}}%
              </div>
            </mat-list-item>
            <mat-divider [inset]="true"></mat-divider>
          </div>
        </ng-container>
        <div *ngIf="!variants.length" class="empty-table-text">
          No records found
          <mat-divider [inset]="true"></mat-divider>
        </div>

        <a [routerLink]="'genetic-variation/high-throughput-sequencing'">
          <mat-card-title class="text-center text-sm-left mt-2 mb-4">
            <h5>Variants of Concern</h5>
          </mat-card-title>
          <div class="row">
            <div class="col-12">
              <app-highcharts
                [chartType]="chartType"
                [chart]="chartOptions"
                [height]="300"
                [hideLegend]="true"
                [series]="chartSeries"
                [xAxisData]="chartData.total.xAxis">
              </app-highcharts>
            </div>
          </div>
        </a>
        <mat-divider></mat-divider>
      </mat-list>
    </mat-card-content>
    <div class="col-xl-12 more-link">
      <a [routerLink]="'genetic-variation/high-throughput-sequencing'">All genetic variation data</a>
    </div>
  </div>
</mat-card>
